srml/contracts/src/rent.rs

```diff
@@ -95,14 +95,26 @@ fn try_evict_or_and_pay_rent<T: Trait>(
 	// The minimal amount of funds required for a contract not to be evicted.
 	let subsistence_threshold = T::Currency::minimum_balance() + <Module<T>>::tombstone_deposit();
 
+	if balance < subsistence_threshold {
+		// The contract cannot afford to leave a tombstone, so remove the contract info altogether.
+		<ContractInfoOf<T>>::remove(account);
+		runtime_io::kill_child_storage(&contract.trie_id);
+		return RentOutcome::Evicted;
+	}
+
 	let dues = fee_per_block
 		.checked_mul(&blocks_passed.saturated_into::<u32>().into())
 		.unwrap_or(<BalanceOf<T>>::max_value());
-
-	let dues_limited = dues.min(contract.rent_allowance);
-	let rent_allowance_exceeded = dues > contract.rent_allowance;
-	let is_below_subsistence = balance < subsistence_threshold;
-	let go_below_subsistence = balance.saturating_sub(dues_limited) < subsistence_threshold;
+	let rent_budget = contract.rent_allowance.min(balance - subsistence_threshold);
+	let go_below_subsistence = rent_budget < dues;
```

Comment:
is this right ? go_below_subsistence is when the balance of the account will go below subsistence after paying dues,

this boolean now represent that a contract got its rent_budget overflowed. If this is what you mean then rename the variable to go_above_rent_budget.

Reply:
Yeah, `go_below_subsistence` is a bad variable name. Should be changed to something like `cannot_pay_rent`, `insufficient_rent`, `must_evict`, something like that...